### ISSUE TITLE
Allow application-defined ephemeral key

### ISSUE THREAD

**Issue by BSipos-RKF** [TimothyClaeys/pycose#50]
The new API for encoding with multiple recipients is much improved from the old API.

Currently when using an algorithm which itself requires an ephemeral sender key (e.g., `EcdhEsA256KW`) the recipient encoding disallows an external (application-defined) ephemeral key with the error:
```
cose.exceptions.CoseException: Unrelated ephemeral public key found in COSE message header
```
This seems overly-strict and disallows deterministic testing. Could this guard be relaxed so that an application could generate its own ephemeral key? The guard can still check that it's the correct key type, curve, etc.

Also, the `CoseRecipient._setup_ephemeral_key()` function generates a COSE key with an default-empty `key_ops` member. The `cose` library doesn't seem to care that the key_ops is empty but it seems like it should either be absent or initialized to include `DeriveKeyOp`.

**Comment by TimothyClaeys** [comment]
Could you provide an example of what you are trying to achieve? In this [example](https://pycose.readthedocs.io/en/latest/examples.html#cose-encrypt) the ephemeral key is set manually. Is this what you are looking for or am I missing something? 

Concerning the empty `key_ops` attribute in the COSE keys, this was also solved by pr #47. When the key ops list is empty it is skipped during the CBOR-encoding (since it is an optional COSE key attribute).

**Comment by BSipos-RKF** [comment]
That example was what I needed. I didn't realize the application needed to inject both the private-key `key` member (for the key agreement) and the public-key header value (for the encoding).
